homeassistant/helpers/entity.py

```diff
@@ -258,6 +263,23 @@ def async_update_ha_state(self, force_refresh=False):
                 temp = units.temperature(float(state), unit_of_measure)
                 state = str(round(temp) if prec == 0 else round(temp, prec))
                 attr[ATTR_UNIT_OF_MEASUREMENT] = units.temperature_unit
+            elif (unit_of_measure in (LENGTH_MILES, LENGTH_YARD, LENGTH_FEET,
```

Comment:
I think the previous conclusion was that the logic that decides what conversion should take place, should be done in the unit system util, ie not here.

Reply:
Thanks Martin, I'll take a look. I already read the comments on #8937 but didn't have time to find a new place.

Reply:
That would be a cool feature for 0.60 / I hope you find time to finish it